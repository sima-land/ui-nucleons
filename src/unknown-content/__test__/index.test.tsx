import { render } from '@testing-library/react';
import { UnknownContent } from '..';

describe('UnknownContent', () => {
  it('should handle no content', () => {
    const { container } = render(<UnknownContent />);

    expect(container.querySelector('div')?.innerHTML).toBe('');
  });

  it('should renders properly', () => {
    const markup = `
      <h2>Для физических лиц</h2>

      <p>
        Оптовая цена вступает в силу, когда в «Корзине» стоимость товаров с этим значком превышает 10&nbsp;000 рублей.
      </p>

      <h2>Для юридических лиц и ИП</h2>

      <p>
        Те же условия + дополнительная скидка при покупке:
      </p>

      <table>
        <thead>
          <tr>
            <td>Сумма товаров</td>
            <td>Скидка от оптовой цены</td>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>От 20 000 ₽</td>
            <td>до 3 %</td>
          </tr>
          <tr>
            <td>От 50 000 ₽</td>
            <td>до 5 %</td>
          </tr>
          <tr>
            <td>От 100 000 ₽</td>
            <td>до 8 %</td>
          </tr>
        </tbody>
      </table>

      <p>
        На часть позиций со значком Опт «Галантерея и швейная галантерея» дополнительные скидки не распространяются.
      </p>
    `;

    const { container } = render(<UnknownContent markup={markup} />);

    expect(container.innerHTML).toMatchSnapshot();
  });

  it('should handle "children" prop', () => {
    const { container } = render(
      <UnknownContent>
        <h1>Primary title</h1>

        <p>Lorem ipsum dolor sit amet consectetur adipisicing elit. Hic, blanditiis!</p>

        <h2>Secondary title</h2>

        <p>Lorem ipsum dolor sit, amet consectetur adipisicing elit. Officiis, ullam?</p>
      </UnknownContent>,
    );

    expect(container.innerHTML).toMatchSnapshot();
  });
});
