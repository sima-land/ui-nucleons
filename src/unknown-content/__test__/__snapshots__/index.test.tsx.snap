// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnknownContent should handle "children" prop 1`] = `"<div class="root"><h1>Primary title</h1><p>Lorem ipsum dolor sit amet consectetur adipisicing elit. Hic, blanditiis!</p><h2>Secondary title</h2><p>Lorem ipsum dolor sit, amet consectetur adipisicing elit. Officiis, ullam?</p></div>"`;

exports[`UnknownContent should renders properly 1`] = `
"<div class="root">
      <h2>Для физических лиц</h2>

      <p>
        Оптовая цена вступает в силу, когда в «Корзине» стоимость товаров с этим значком превышает 10&nbsp;000 рублей.
      </p>

      <h2>Для юридических лиц и ИП</h2>

      <p>
        Те же условия + дополнительная скидка при покупке:
      </p>

      <table>
        <thead>
          <tr>
            <td>Сумма товаров</td>
            <td>Скидка от оптовой цены</td>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td data-table-column-title="Сумма товаров">От 20 000 ₽</td>
            <td data-table-column-title="Скидка от оптовой цены">до 3 %</td>
          </tr>
          <tr>
            <td data-table-column-title="Сумма товаров">От 50 000 ₽</td>
            <td data-table-column-title="Скидка от оптовой цены">до 5 %</td>
          </tr>
          <tr>
            <td data-table-column-title="Сумма товаров">От 100 000 ₽</td>
            <td data-table-column-title="Скидка от оптовой цены">до 8 %</td>
          </tr>
        </tbody>
      </table>

      <p>
        На часть позиций со значком Опт «Галантерея и швейная галантерея» дополнительные скидки не распространяются.
      </p>
    </div>"
`;
